src/uucore/src/lib/features/entries.rs

```diff
@@ -72,6 +72,31 @@ pub fn get_groups() -> IOResult<Vec<gid_t>> {
     }
 }
 
+pub fn get_groups_gnu(arg_id: Option<u32>) -> IOResult<Vec<gid_t>> {
+    // The list of group IDs return from GNU's `groups` and GNU's `id --groups`
+    // starts with the effective group ID (egid).
+    // This is a wrapper for `get_groups()` to mimic this behavior.
+    //
+    // If `arg_id` is None (default), `get_groups_gnu` moves the effective
+    // group id (egid) to the first entry in the returned Vector.
+    // If `arg_id` is Some(x), `get_groups_gnu` moves the id with value `x`
+    // to the first entry in the returned Vector. This might be necessary
+    // for `id --groups --real` if `gid` and `egid` are not equal.
+
+    let mut groups = get_groups()?;
+    if groups.len() > 1 {
+        let egid = arg_id.unwrap_or_else(|| crate::features::process::getegid());
+        if *groups.first().unwrap() != egid {
+            if let Some(index) = groups.iter().position(|&x| x == egid) {
+                groups.remove(index);
+                groups.insert(0, egid);
+            }
+        }
+    }
+    Ok(groups)
+}
+
+#[derive(Copy, Clone)]
```

Comment:
What was your reason for adding this?

Reply:
Without this, there's a "use of moved value: `possible_pw` value used here after move" in L205 (and L199).